chalice_spec/chalice.py

```diff
@@ -38,8 +44,13 @@ def route(path: str, **kwargs):
             :param kwargs: Additional Chalice kwargs and APIspec definitions.
             """
             docs: Docs = kwargs.pop("docs", None)
+            methods = [method.lower() for method in kwargs.get("methods", ["get"])]
+
+            if docs is None and self._generate_default_docs:
+                docs = Docs(**{method: Operation(response=BaseModel, request=(None if method in ['get', 'delete', 'head', 'options'] else BaseModel)) for method in methods})
```

Comment:
This looks like a neat addition! I see you've added `patch` to the methods list in `docs.py`. Would you like to also add `patch` to this list?

Reply:
This is setting the request schema to `None` when the method is in this list (because we don't expect a request body for `GET`, `DELETE` methods etc). So because we'd expect a `PATCH` to have a request body, I don't think we want it in this list.
I appreciate that this line is pretty messy, I tried to reformat it and it didn't really help. Open to suggestions.